File: frontend/html/scripts/protocol/server/messages/ServerMessageHandshake.js
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";

export default class ServerMessageHandshake extends ServerMessage {
    constructor(reader) {
        super(ServerMessageType.HANDSHAKE);
        this.id = reader.readBigInt64();
        this.token = reader.readBigInt64();
    }
}

File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/server/messages/ServerMessageHandshake.java
package net.stzups.board.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.PersistentUserSession;
import net.stzups.board.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageHandshake extends ServerMessage {
    private long id;
    private long token;

    public ServerMessageHandshake(PersistentUserSession persistentUserSession) {
        super(ServerMessageType.HANDSHAKE);
        this.id = persistentUserSession.getId();
        this.token = persistentUserSession.generateToken();
    }

    @Override
    public void serialize(ByteBuf bytebuf) {
        super.serialize(bytebuf);
        bytebuf.writeLong(id);
        bytebuf.writeLong(token);
    }
}
